cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs

```diff
@@ -122,20 +105,25 @@ class
 sizeHash :: forall h proxy. HashAlgorithm h => proxy h -> Word
 sizeHash _ = fromInteger (natVal (Proxy @(SizeHash h)))
 
-newtype Hash h a = UnsafeHashRep (HashRep h)
+newtype Hash h a = UnsafeHashRep (PackedBytes (SizeHash h))
 
-deriving instance HashAlgorithm h => Eq (Hash h a)
-deriving instance HashAlgorithm h => Ord (Hash h a)
-deriving instance HashAlgorithm h => NFData (Hash h a)
-deriving instance HashAlgorithm h => NoThunks (Hash h a)
 deriving instance Generic (Hash h a)
+deriving via OnlyCheckWhnfNamed "Hash" (Hash h a) instance NoThunks (Hash h a)
 
-pattern UnsafeHash :: forall h a. HashAlgorithm h => ShortByteString -> Hash h a
-pattern UnsafeHash bytes <- UnsafeHashRep (fromHashRep (Proxy :: Proxy h) -> bytes)
-  where
-  UnsafeHash bytes = UnsafeHashRep (unsafeToHashRep (Proxy :: Proxy h) bytes)
+instance NFData (Hash h a) where
+  rnf = rwhnf
+
+instance KnownNat (SizeHash h) => Eq (Hash h a) where
```

Comment:
I thought the purpose of the change was to no longer have a constraint bubbling up from the Eq instance.